kafka-producers/templates/deployment.yaml

```diff
@@ -0,0 +1,68 @@
+{{- range $producer, $values := .Values.producers }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ printf "%s-%s" $.Release.Name $producer | trunc 63 | trimSuffix "-" }}
+  labels:
+    helm.sh/chart: {{ include "kafka-producers.chart" $ }}
+    app.kubernetes.io/instance: {{ $.Release.Name }}
+  {{- include "multicast.annotation" . | nindent 2 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ printf "%s-%s" $.Release.Name $producer | trunc 63 | trimSuffix "-" }}
+      app.kubernetes.io/instance: {{ $.Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ printf "%s-%s" $.Release.Name $producer | trunc 63 | trimSuffix "-" }}
+        app.kubernetes.io/instance: {{ $.Release.Name }}
+      {{- include "multicast.annotation" . | nindent 6 }}
+    spec:
+      containers:
+        - name: {{ printf "%s-%s" $.Release.Name $producer | trunc 63 | trimSuffix "-" }}
+          {{- if .image }}
+          {{- $image := .image.repository | default $.Values.defaults.image.repository }}
+          {{- $tag := .image.tag | default $.Values.defaults.image.tag}}
+          image: "{{ $image }}:{{ $tag }}"
+          {{- else }}
+          image: "{{ $.Values.defaults.image.repository }}:{{ $.Values.defaults.image.tag }}"
+          {{- end }}
+          {{- if .image }}
+          {{- $pullPolicy := .image.pullPolicy | default $.Values.defaults.image.pullPolicy }}
+          imagePullPolicy: {{ $pullPolicy }}
+          {{- else }}
+          imagePullPolicy: {{ $.Values.defaults.image.pullPolicy }}
+          {{- end }}
+          {{- include "kafka-producers.container-boilerplate" $ | nindent 10 }}
```

Comment:
I don't think you want this boilerplate line here, you want it below on line 47 - although since there should I think be just one place where you call it maybe just put it right in this file and get rid of the helpers part?